### PR TITLE
qt: Remove redundant stopThread() and stopExecutor() signals

### PR DESCRIPTION
**Pull request by hebasto**
The `QThread::finished` signal do this work.

### CODE REVIEW

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -379,8 +378,7 @@ void BitcoinApplication::startThread()
     connect(this, &BitcoinApplication::requestedInitialize, executor, &BitcoinCore::initialize);
     connect(this, &BitcoinApplication::requestedShutdown, executor, &BitcoinCore::shutdown);
     /*  make sure executor object is deleted in its own thread */
-    connect(this, &BitcoinApplication::stopThread, executor, &QObject::deleteLater);
-    connect(this, &BitcoinApplication::stopThread, coreThread, &QThread::quit);
+    connect(coreThread, &QThread::finished, executor, &QObject::deleteLater);
```
[the remark this answers is not in the document]
Comment (hebasto) [reply src/qt/bitcoin.cpp]: @promag https://doc.qt.io/qt-5.9/qthread.html:
> From Qt 4.8 onwards, it is possible to deallocate objects that live in a thread that has just ended, by connecting the finished() signal to QObject::deleteLater().

https://doc.qt.io/qt-5.9/qthread.html#finished

[the remark this answers is not in the document]
Comment (promag) [reply src/qt/bitcoin.cpp]: TIL.

**Comment by jonasschnelli** [comment]
seems more elegant, utACK 041be348b0bd96d87eb3d1a2b0edc47538dc9f66 will test soon.

**Comment by jonasschnelli** [comment]
Tested ACK #041be348b0bd96d87eb3d1a2b0edc47538dc9f66
https://bitcoin.jonasschnelli.ch/build/795

**Comment by hebasto** [comment]
@Sjors Could you possibly to review this PR?

**Comment by jonasschnelli** [comment]
Needs more tests (@Sjors, @fanquake, etc.)...

**Comment by fanquake** [comment]
tACK 041be34

Tested on top of 816fab9. If anyone has extra tests etc that could be run against this please post them here.

**Comment by hebasto** [comment]
Added similar cleanup for `RPCExecutor`. Please re-review.

**Review by ken2812221 — APPROVED** [review]
tACK 4512ac8bdec9446674e774a83164419caae4b4ed

**Comment by MarcoFalke** [comment]
Concept ACK

**Comment by hebasto** [comment]
Rebased.

**Comment by hebasto** [comment]
Rebased. Again :)

**Comment by laanwj** [comment]
utACK 24313fbf7e3d69145bc18c089601ba7aea35d61c

**Pull request merged**